base/doc/usrguide.tex

```diff
@@ -1161,7 +1161,7 @@ \section{Expandable floating point (and other) calculations}
 above. Nevertheless, in practice they are usually sufficient.  For
 example
 \begin{verbatim}
-\newcommand\calculateheight[1]{%
+\NewDocumentCommand\calculateheight{m}{%
```

Comment:
My personal opinion is that as long as the "m" only case of \NewDocumentCommand isn't streamlined \newcommand does have its place (also when expansion is wanted) because the \NewDocumentMachinery is rather heavy when ending up with essentialls \protected\def in the end.  But this is kind of a different topic and fine with promoting \NewDocumentCommand here.

Reply:
@FrankMittelbach We could bring back shortcutting for all-`m` arguments

Reply:
did we ever had it, thought we only talked about it? But yes, I think it makes a lot of sense because it cuts down on the run time costs a lot moving them to the declaration.

Reply:
> did we ever had it, thought we only talked about it? But yes, I think it makes a lot of sense because it cuts down on the run time costs a lot moving them to the declaration.

We had something: I'll log an issue and put in the pool for 2024-06-01.

Reply:
Logged as https://github.com/latex3/latex2e/issues/1189